docs/dev/go-style-guide.md

```diff
@@ -0,0 +1,116 @@
+
+``` golang
+type middleware struct {
+	middleware Middleware
+}
+```
+
+- Acronyms are all capitalized, like "RPC", "gRPC", "API". "MyID", rather than "MyId".
+- Whenever it is safe to use Go's built-in `error` instantiation functions (as opposed to Cosmos SDK's error instantiation functions), prefer `errors.New()` instead of `fmt.Errorf()` unless you're actually using the format feature with arguments.
+
+## Importing libraries
+
+- Use [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports).
+- Separate imports into blocks: one for the standard lib, one for external libs and one for application libs. For example:
+
+	```golang
+	import (
+		// standard library imports
+		"fmt"
+		"testing"
+					
+		// external library imports
+		"github.com/stretchr/testify/require"
+		abci "github.com/tendermint/tendermint/abci/types"
+					
+		// ibc-go library imports
+		"github.com/cosmos/ibc-go/modules/core/23-commitment/types"
+	)
+	```
+
+## Dependencies
+
+- Dependencies should be pinned by a release tag, or specific commit, to avoid breaking `go get` when external dependencies are updated.
+- Refer to the [contributing](./development.md#dependencies) document for more details.
+
+## Testing
+
+- Make use of table driven testing where possible and not-cumbersome. Read [this blog post](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go) for more information. See the [tests](https://github.com/cosmos/ibc-go/blob/f24f41ea8a61fe87f6becab94e84de08c8aa9381/modules/apps/transfer/keeper/msg_server_test.go#L11) for [`Transfer`](`https://github.com/cosmos/ibc-go/blob/f24f41ea8a61fe87f6becab94e84de08c8aa9381/modules/apps/transfer/keeper/msg_server.go#L15) for an example.
+- Make use of Testify [assert](https://godoc.org/github.com/stretchr/testify/assert) and [require](https://godoc.org/github.com/stretchr/testify/require).
+- When using mocks, it is recommended to use Testify [mock](https://pkg.go.dev/github.com/stretchr/testify/mock) along with [Mockery](https://github.com/vektra/mockery) for autogeneration.
+
+## Errors
+
+- Ensure that errors are concise, clear and traceable.
+- Depending on the context, use either `cosmossdk.io/errors` or `stdlib` error packages.
+- For wrapping errors, use `fmt.Errorf()` with `%w`.
+- Panic is appropriate when an internal invariant of a system is broken, while all other cases (in particular, incorrect or invalid usage) should return errors.
+ke it is a special case with the SDK where returning errors can sometimes reset state changes and panics are caught and as a result will not result in the shutdown of the system.
+- Error messages should be formatted as following:
+
+	```golang
+	sdkerrors.Wrapf(
+		<most specific error type possible>,
+		"<optional text description ended by colon and space>expected %s, got %s",
+		<value 1>,
+		<value 2>
+	)
+	```
```

Comment:
Seems like the formatting of this is a little off? Sentence starts with "ke", should this be "Take"?

Reply:
It's indented to the right a bit so that aligns with the text in the bullet point that it belongs to.